x/halo/keeper/keeper.go

```diff
@@ -48,8 +51,28 @@ func (k *Keeper) SetBankKeeper(bankKeeper types.BankKeeper) {
 	k.bankKeeper = bankKeeper
 }
 
+// SetFTFKeeper overwrites the fiattokenfactory keeper used in this module.
+func (k *Keeper) SetFTFKeeper(ftfKeeper types.FiatTokenFactoryKeeper) {
+	k.ftfKeeper = ftfKeeper
+}
+
 // SendRestrictionFn executes necessary checks against all USYC transfers.
 func (k *Keeper) SendRestrictionFn(ctx sdk.Context, fromAddr, toAddr sdk.AccAddress, amt sdk.Coins) (newToAddr sdk.AccAddress, err error) {
+	denom := k.ftfKeeper.GetMintingDenom(ctx).Denom
+	if amount := amt.AmountOf(denom); !amount.IsZero() {
```

Comment:
At this point I would suggest replacing `IsZero()` with `IsPositive()` also here
```suggestion
	if amount := amt.AmountOf(denom); !amount.IsPositive() {
```

Reply:
I think this is fine given that the `x/bank` module has checks in place for negative values.